autogtp/Job.cpp

```diff
@@ -158,25 +161,27 @@ Result ValidationJob::execute(){
     QString wmove = "play white ";
     QString bmove = "play black ";
     do {
-        first.move();
-        if (!first.waitForMove()) {
-            return res;
-        }
-        first.readMove();
-       m_boss->incMoves();
-        if (first.checkGameEnd()) {
-            break;
+        if (first.getToMove() == Game::BLACK) {
+            first.move();
+            if (!first.waitForMove()) {
+                return res;
+            }
+            first.readMove();
+            m_boss->incMoves();
+            if (first.checkGameEnd()) {
+                break;
+            }
+            second.setMove(bmove + first.getMove());
+            first.nextMove();
         }
-        second.setMove(bmove + first.getMove());
         second.move();
         if (!second.waitForMove()) {
             return res;
         }
         second.readMove();
-       m_boss->incMoves();
+        m_boss->incMoves();
         first.setMove(wmove + second.getMove());
-        second.nextMove();
-    } while (first.nextMove() && m_state.load() == RUNNING);
+    } while (second.nextMove() && m_state.load() == RUNNING);
```

Comment:
`second.nextMove()` is needed to keep the right Black White order. Or do you see a cool optimization I don't?

Reply:
It is still there :) I’ve moved the call for first further up, keeping it with the rest of the stuff for black’s move, and put the call for second in the while test